Cython/Compiler/PyrexTypes.py

```diff
@@ -3799,10 +3806,14 @@ def create_from_py_utility_code(self, env):
                 'maybe_unordered': self.maybe_unordered(),
                 'type': self.cname,
             })
+            directives = dict(env.directives)
+            # if present, cpp_locals will invalidate some utility code
+            # (TODO filter directives more thoroughly to keep only those that are relevant)
+            directives["cpp_locals"] = False
```

Comment:
I hadn't noticed that we're doing this in two places.
```suggestion
            # Override directives that should not be inherited from user code.
            # TODO: filter directives with an allow list to keep only those that are safe and relevant.
            directives = dict(env.directives, cpp_locals=False)```